pycparser/c_parser.py

```diff
@@ -973,47 +1011,43 @@ def p_enumerator(self, p):
 
         p[0] = enumerator
 
-    def p_declarator_1(self, p):
-        """ declarator  : direct_declarator
+    def p_declarator(self, p):
+        """ declarator  : id_declarator
+                        | typeid_declarator
         """
         p[0] = p[1]
 
-    def p_declarator_2(self, p):
-        """ declarator  : pointer direct_declarator
+    @parameterized(('id', 'ID'), ('typeid', 'TYPEID'), ('typeid_noparen', 'TYPEID'))
+    def _p_XXX_declarator_1(self, p):
```

Comment:
Ah, OK. That makes sense. I don't think the extra complexity is worth it (having a metaclass or class decorator here), so we can keep it as is.